### PR TITLE
Use threadlocal storage for pooling our matrices.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @jasonmalinowski

File: src/Workspaces/Core/Portable/Utilities/EditDistance.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 using System.Text;
 using Microsoft.CodeAnalysis;
 using static System.Math;
+using System.Threading;
```
Comment (DustinCampbell) [inline src/Workspaces/Core/Portable/Utilities/EditDistance.cs]: Sort

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
test prtest/win/vsi/p0  please

**Comment by CyrusNajmabadi** [comment]
test prtest/win/vsi/p2  please

**Comment by CyrusNajmabadi** [comment]
retest prtest/win/vsi/p0 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/win/vsi/p0 please

**Comment by jmarolf** [comment]
:+1:

**Pull request merged**